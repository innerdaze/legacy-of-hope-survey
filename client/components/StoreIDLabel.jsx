import React from 'react'
import PropTypes from 'prop-types'
import Label from 'grommet/components/Label'

const StoreIDLabel = ({ storeID }) =>
  <Label>
    StoreID: {storeID}
  </Label>

StoreIDLabel.propTypes = {
  storeID: PropTypes.string.isRequired
}

export default StoreIDLabel
