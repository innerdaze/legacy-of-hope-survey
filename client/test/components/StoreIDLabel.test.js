import React from 'react'
import StoreIDLabel from '../../components/StoreIDLabel'

let wrapper

describe('Testing on StoreIDLabel...', () => {
  beforeEach(() => {
    wrapper = shallow(<StoreIDLabel storeID='FooID' />)
  })

  describe('StoreIDLabel layout', () => {
    test('Expect to have a Label', () => {
      expect(wrapper.find('Label')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect to have a Label similar with prop value passed in', () => {
      let storeID = 'FooID'
      wrapper = render(<StoreIDLabel storeID={storeID} />)

      expect(wrapper.text()).toContain(storeID)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
