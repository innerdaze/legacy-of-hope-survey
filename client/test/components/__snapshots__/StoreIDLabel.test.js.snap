// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on StoreIDLabel... StoreIDLabel layout Expect to have a Label 1`] = `
<Label
  size="medium"
>
  StoreID: 
  FooID
</Label>
`;

exports[`Testing on StoreIDLabel... Test on prop Expect to have a Label similar with prop value passed in 1`] = `
<label
  class="grommetux-label grommetux-label--margin-medium grommetux-label--medium"
>
  StoreID: FooID
</label>
`;
